### DISCUSSION TITLE
advanced "inner join" or "where in subselect" queries

### DISCUSSION THREAD

**Discussion by shainegordon** [prisma/prisma#2836] [Q&A] [answered]
Is it possible, without using `rawQuery` to do an `inner join`, or `where in subselect`.

best illustrated with some SQL queries using a theoretical data model:

```
SELECT *
FROM "OrderItem"
WHERE "productId" IN
      (SELECT "productId" FROM "ShoppingCartItem" WHERE "customerId" = '1');
```
OR 
```
SELECT "OrderItem".*
FROM "OrderItem"
         INNER JOIN "ShoppingCartItem" ON "Order"."productId" = "ShoppingCartItem"."productId"
WHERE "customerId" = '1';
```
both these queries return the same thing

**Comment by TomasMoskala** [comment]
I think you can define custom field resolver in your object
https://www.youtube.com/watch?v=IiBED__5k1Y
or use foreign keys to connect the tables

**Reply by shainegordon** [reply]
Sorry, I should have been clear.

In this instance I am using Prisma as an ORM of sorts, not binding it with Nexus.

So I want to load some data with prisma, and then use that data in my backend

**Reply by TomasMoskala** [reply]
I am not much experienced but my Prisma models generate additional array fields whenever there is a foreign key reference. It should be similar.

**Comment by ryands17** [comment, marked as answer]
Hey @shainegordon :wave: 
Currently I don't think this is possible without a raw query. Could you please create a feature request [here](https://github.com/prisma/prisma/discussions/2834) regarding the above use case.

**Comment by ludralph** [comment]
Hi @shainegordon,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
